nx/lib/nx.ex

```diff
@@ -1573,21 +1573,54 @@ defmodule Nx do
     impl!(tensor).to_batched_list(%{tensor | shape: put_elem(shape, 0, batch_size)}, tensor, opts)
   end
 
+  @doc """
+  Returns the underlying tensor as a number.
+
+  If the tensor has a dimension, it raises.
+
+  ## Examples
+
+      iex> Nx.to_number(1)
+      1
+
+      iex> Nx.to_number(Nx.tensor([1.0, 2.0, 3.0]))
+      ** (ArgumentError) cannot convert tensor of shape {3} to number
+
+  """
+  @doc type: :conversion
+  def to_number(tensor)
+
+  def to_number(number) when is_number(number), do: number
+
+  def to_number(tensor) do
+    tensor = to_tensor(tensor)
+
+    if tensor.shape != {} do
+      raise ArgumentError, "cannot convert tensor of shape #{inspect(tensor.shape)} to number"
+    end
+
+    match_types [tensor.type] do
+      <<match!(x, 0)>> = to_binary(tensor)
+      read!(x, 0)
+    end
+  end
+
   @doc """
   Returns the underlying tensor as a scalar.
 
   If the tensor has a dimension, it raises.
 
   ## Examples
 
-      iex> Nx.to_scalar(1)
+      iex> Nx.to_number(1)
       1
 
-      iex> Nx.to_scalar(Nx.tensor([1.0, 2.0, 3.0]))
-      ** (ArgumentError) cannot convert tensor of shape {3} to scalar
+      iex> Nx.to_number(Nx.tensor([1.0, 2.0, 3.0]))
+      ** (ArgumentError) cannot convert tensor of shape {3} to number
 
   """
   @doc type: :conversion
+  @deprecated "Use Nx.to_number/1 instead"
```

Comment:
We don't need to deprecate it, you can go ahead and remove it. :)

Reply:
Axon will not be happy.

Reply:
Ok, let's keep this for a week and then we can change it. :)

Reply:
:) Alright.